utils/InputValidation.tsx

```diff
@@ -168,7 +195,7 @@ export const useValidation = () => {
   return {
     email: {
       value: email,
-      isValid: isEmailValid,
+      isValid: isEmailValid && !serverErrors.email,
       handleChange: handleEmailChange,
       handleBlur: handleEmailBlur,
```

Comment:
원래 작성해주셨던 함수가 훨씬 깔끔하긴 하지만
반복되는 함수들을 공통화하는 것도 좋을 거 같아요.

예를 들어서
```suggestion
      handleBlur: () =>
        handleBlur(
          email.value,
          /^[^\s@]+@[^\s@]+\.[^\s@]+$/,
          (valid) => setEmail((prev) => ({ ...prev, isValid: valid })),
        ),
```
```
  const handleBlur = (value: string, regex: RegExp | null, setValid: (valid: boolean) => void) => {
  setValid(value !== '' && (regex ? regex.test(value) : true));
  };
```
이런 식으로도 가능할 거 같긴 합니다. 참고만 해주세요~!!